packages/manager/modules/anthos/src/dashboard/components/assign-private-ip/assign-private-ip.controller.js

```diff
@@ -0,0 +1,125 @@
+import { IPV4_REGEX } from './constant';
+
+export default class AssignPrivateIpController {
+  /* @ngInject */
+  constructor($translate, Alerter, AnthosTenantsService) {
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.AnthosTenantsService = AnthosTenantsService;
+
+    this.ipModel = {
+      form: null,
+      labels: {
+        privateInputIp: 'privateInputIp',
+      },
+      validationKeys: {
+        INVALID_IP_FORMAT: 'INVALID_IP_FORMAT',
+        INVALID_PRIVATE_IP: 'INVALID_PRIVATE_IP',
+        INVALID_RANGE_IP: 'INVALID_RANGE_IP',
+      },
+      privateIp: {
+        range: null,
+      },
+    };
+  }
+
+  static isValidV4Ip(ip) {
+    return IPV4_REGEX.IP_V4_FORMAT.test(ip);
+  }
+
+  static isValidCidrIp(range) {
+    return IPV4_REGEX.IP_CIDR_V4_FORMAT.test(range);
+  }
+
+  static isValidRangeIp(range) {
+    const Ctrl = AssignPrivateIpController;
```

Comment:
Why not access with `this` ?

Reply:
Both are equivalent (same reference).
I preferred this notation because it's shortest than to write `this.constructor.XXXX` static functions.

Reply:
One small thing, may be you can add this in `$onInit` to avoid multiple declarations in functions?

Reply:
Static function are always better to accesses with Class Name within the same file. "this.constructor.XXXX" should only be used outside the file when class is not available and instance is available.